unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/GotoPanelHUD/GotoPanelHUDView.cs

```diff
@@ -0,0 +1,96 @@
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+using DCL.Interface;
+using DCL;
+using System;
+
+public class GotoPanelHUDView : MonoBehaviour
+{
+    [SerializeField] private Button teleportButton;
+    [SerializeField] private TextMeshProUGUI panelText;
+    [SerializeField] public GameObject container;
+    [SerializeField] internal TextMeshProUGUI sceneTitleText;
+    [SerializeField] internal TextMeshProUGUI sceneOwnerText;
+    [SerializeField] internal Sprite scenePreviewFailImage;
+    [SerializeField] internal RawImageFillParent scenePreviewImage;
+    [SerializeField] internal GameObject loadingSpinner;
+    [SerializeField] internal Button closeButton;
+    [SerializeField] internal Button cancelButton;
+    [SerializeField] internal ShowHideAnimator contentAnimator;
+
+    private ParcelCoordinates targetCoordinates;
+
+    AssetPromise_Texture texturePromise = null;
+
+    private void Start()
+    {
+        teleportButton.onClick.RemoveAllListeners();
+        teleportButton.onClick.AddListener(TeleportTo);
+        closeButton.onClick.AddListener(OnClosePressed);
+        cancelButton.onClick.AddListener(OnClosePressed);
+        container.SetActive(false);
+        contentAnimator.OnWillFinishHide += (animator) => Hide();
+    }
+
+    private void TeleportTo()
```

Comment:
I would have this logic in the controller. The view would just raise an event `OnTeleportClicked`.